backend/clubs/views.py

```diff
@@ -6700,26 +6700,25 @@ def list(self, *args, **kwargs):
         """
 
         app_id = self.kwargs["application_pk"]
+        serializer = kwargs.get("serializer", self.get_serializer_class())
         key = f"applicationsubmissions:{app_id}"
 
         cached = cache.get(key)
         if cached is not None:
-            return Response(cached)
+            return Response(serializer(cached, many=True).data)
         else:
-            serializer = self.get_serializer_class()
             qs = self.get_queryset()
+            cache.set(key, qs, 60 * 60)
```

Comment:
Is there a reason we're caching the queryset instead of the data?

Reply:
We only get the data after running the queryset through the serializer, while I wanted the cache contents itself to be serializer agnostic given that export uses a different serializer. On second thought though, get_queryset() might be evaluated lazily so I would need to rethink this.

Reply:
Do you think .objects.all() would work @aviupadhyayula?

Reply:
[From what I've read](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://betterprogramming.pub/understanding-django-database-querysets-and-its-optimizations-1765cb9c36e5%23:~:text%3DPickling%252FCaching,of%2520model%2520objects%2520from%2520cache.&ved=2ahUKEwi4zYmj-biIAxWEEFkFHQbLCCEQFnoECBYQAw&usg=AOvVaw25zhDELVWq2rqCDrZSCpg7), it looks like the Django caching library forces evaluation upon insertion. We can test this later (force Django to log each query evaluation in settings). I'm more concerned about breaking our pattern of not caching querysets anywhere.

Honestly, is there a reason to cache this view? Querying the list of applicants will be done very rarely. Some latency on invocation is preferable to leaving a couple applicants off the UI or downloaded CSV.

Reply:
That's fair, caching definitely isn't necessary, at least for the CSV export (I didn't know that clubs used this feature until very recently). However, the cache invalidation logic was already previously written for the list view, and application-related views also contribute to site-level performance (I'm not sure how strict our transaction isolation levels are if at all, but you could imagine that querying all applications repeatedly, especially while new upsert transactions could still occur, is bad during application season).